### DISCUSSION TITLE
How to reference component?

### DISCUSSION THREAD

**Discussion by wlkns** [alpinejs/alpine#776] [4. General]
I see that $el gives me the root element, what if I want to access the component e..g $self ?

I'm trying to conditionally disable checkboxes that are not checked when a certain amount are checked.

```
<input type="checkbox" v-model="categories" value="1" x-bind:disabled="categories.indexOf($self.value) === -1 && categories.length > 5" /> 
```

Any help would be appreciated.

**Comment by ryangjchandler** [comment]
@wlkns Alpine doesn't provide anything like that out of the box.

The best way to do it would be using `x-ref` and `$refs`.

**Reply by wlkns** [reply]
Ah, unfortunate, I'm sure I could find a few use cases.

I did consider using x-for with x-ref to compare the values, I also had hard coded values through PHP working.

Thanks for your reply!

**Comment by MichaelPote** [comment]
Yes, this is still needed in 2023.

    <div id="page-header" x-data="{
        isAtTop: true
        onWindowScroll: () => { isAtTop = window.scrollY <= 5 },
    }"
     x-on:scroll.window.debounce.200="onWindowScroll"
     x-on:scroll.window.throttle="onWindowScroll"
    >

In the above, isAtTop will never update because `onWindowScroll` will always be called from the `Window`'s context.

**Reply by ekwoka** [reply]
In this case, it's your code being wrong.

```
x-data="{
    isAtTop: true
    onWindowScroll(){ this.isAtTop = window.scrollY <= 5 },
}"
```

would work

but, this magic does exist: `$data`

**Reply by MichaelPote** [reply]
Ah okay I didnt notice the difference between `onWindowScroll: () => {}` and `onWindowScroll() {}` Interesting...

**Reply by ekwoka** [reply]
The first only uses the outside scope.

The second will bind `this` which is the data stack.

**Reply by MichaelPote** [reply]
I thought the 'outside scope' would be the context in which the `x-data` array is evaluated, but it's actually the context in which the `onWindowScroll` function is executed.... Cool!

**Reply by ekwoka** [reply]
no, it would be in which x-data is evaluated, which does not include `isAtTop` because `isAtTop` didn't exist in the data stack above that element.

**Reply by MichaelPote** [reply]
Sorry if this is getting into the weeds, but why then when I do: `onWindowScroll: () => { console.log(this); }` I get `Window`?

**Reply by ekwoka** [reply]
yes, because there is no this to bind, so it defaults to the `globalThis`.

But it's evaluated in the context of creating the x-data which would include the datastack OUTSIDE of the x-data, but not the one created by it, since it doesn't exist yet.

**Reply by MichaelPote** [reply]
Ahah okay thanks that makes sense!
